shared/src/main/scala/pl/metastack/metatime/Time.scala

```diff
@@ -113,6 +155,38 @@ object Time {
     override val ms: Int = millisecond
   }
 
+  def apply(comp : Component) : Time = {
+    comp match {
+      case _: Time =>
```

Comment:
I tried to remove casts but compiler cannot resolve .h .m .s .ms symbols at run time and we get compile time error. So unfortunately we will have to live with them, I guess.